hr_holidays_public/models/hr_leave.py

```diff
@@ -19,6 +19,16 @@ def _get_number_of_days(self, date_from, date_to, employee_id):
             date_from, date_to, employee_id
         )
 
+    @api.depends("number_of_days")
+    def _compute_number_of_hours_display(self):
+        if self.holiday_status_id.exclude_public_holidays or not self.holiday_status_id:
+            instance = self.with_context(
+                exclude_public_holidays=True, employee_id=self.employee_id.id
+            )
+        else:
+            instance = self
```

Comment:
```suggestion
    @api.depends("number_of_days")
    def _compute_number_of_hours_display(self):
        context = {}
        if self.holiday_status_id.exclude_public_holidays or not self.holiday_status_id:
            context.update({
                "exclude_public_holidays": True,
                "employee_id": self.employee_id.id,
            })
        instance = self.with_context(**context)
```
This will avoid codecov to scream ^^

Reply:
That's not a big problem in this case, and you will be tricking it a bit, as in fact, we are not covering the case where the holidays type doesn't have the check.